tests/normalization_test.py

```diff
@@ -55,6 +58,81 @@ def test_solid_angle_keeps_relevant_masks():
     assert 'mask2' not in solid_angle.masks
 
 
+@pytest.mark.skipif(
+    'SANS2D00063091.nxs' not in os.listdir(), reason='Needs data to recreate reference'
+)
+@pytest.mark.filterwarnings("ignore:.*")
+def test_recreate_sans2d_solidangle_from_mantid_file():
+    simpleapi = pytest.importorskip("mantid.simpleapi")
+    scippneutron = pytest.importorskip("scippneutron")
+
+    ws = simpleapi.Load('SANS2D00063091.nxs')
+    radius = 0.00405 * sc.Unit('m')
+    length = 0.002033984375 * sc.Unit('m')
+
+    simpleapi.SetInstrumentParameter(
+        ws,
+        ParameterName='x-pixel-size',
+        ParameterType='Number',
+        Value=str((2 * radius).to(unit='mm').value),
+    )
+    simpleapi.SetInstrumentParameter(
+        ws,
+        ParameterName='y-pixel-size',
+        ParameterType='Number',
+        Value=str(length.to(unit='mm').value),
+    )
+
+    outWs = simpleapi.SolidAngle(ws, method='HorizontalTube')
+    da = scippneutron.from_mantid(outWs)['data']['spectrum', :120000:100]
+    fname = os.path.join(
+        os.path.dirname(__file__), 'SANS2D00063091.SolidAngle_from_mantid.hdf5'
+    )
+    da_old = sc.io.load_hdf5(filename=fname)
+    assert sc.identical(da, da_old)
+    # Recreate reference file
+    # sc.io.hdf5.save_hdf5(da, fname)
+
+
+def test_solid_angle():
+    da = sc.io.load_hdf5(
+        filename=os.path.join(
+            os.path.dirname(__file__), 'SANS2D00063091.SolidAngle_from_mantid.hdf5'
+        )
+    )
+    mantid_solid_angle = da.data['tof', 0]
+
+    R = 0.00405
+    L = 0.002033984375
+    pixel_shape = {
+        'vertices': sc.array(
+            dims=['vertex'],
+            values=[
+                [0, 0, 0],
+                [R, 0, 0],
+                [0, L, 0],
+            ],
+            dtype=sc.DType.vector3,
+            unit='m',
+        )
+    }
```

Comment:
You may be looking for https://scipp.github.io/generated/functions/scipp.vectors.html, which simpifies the creation.